keystore/file_keystore_fips.go

```diff
@@ -0,0 +1,84 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build requirefips
+
+package keystore
+
+import (
+	"github.com/elastic/elastic-agent-libs/config"
+)
+
+// NopKeystore is used in FIPS mode to a beats keystore.
+type NopKeystore struct{}
+
+// NewFileKeystoreWithPasswordAndStrictPerms returns a NopKeystore.
+//
+// It is using an exising, exported function call name so callers do not need to change the function call, just the build tag.
+func NewFileKeystoreWithPasswordAndStrictPerms(keystoreFile string, password *SecureString, strictPerms bool) (Keystore, error) {
```

Comment:
Should we have tests for the fips version as well? If the  code here has a bug but we don't test with requirefips we'll only catch it when building beats or elastic-agent with fips enabled.

Reply:
Added tests